### PR TITLE
carthage 0.31.0

### PR DESCRIPTION
**Pull request by jdhealy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
  - Not quite the same, see patch — didn’t want to force push over @blender
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Update Carthage to 0.31.0 · Supersedes homebrew/homebrew-core#32403

Patches `OTHER_LDFLAGS` `headerpad_max_install_names` to avoid:

```
Changing dylib ID of /usr/local/Cellar/carthage/0.30.1/Frameworks/CarthageKit.framework/Versions/A/Frameworks/Tentacle.framework/Versions/A/Tentacle
  from @rpath/Tentacle.framework/Versions/A/Tentacle
    to /usr/local/opt/carthage/Frameworks/CarthageKit.framework/Versions/A/Frameworks/Tentacle.framework/Versions/A/Tentacle
Error: Failed changing dylib ID of /usr/local/Cellar/carthage/0.30.1/Frameworks/CarthageKit.framework/Versions/A/Frameworks/Tentacle.framework/Versions/A/Tentacle
  from @rpath/Tentacle.framework/Versions/A/Tentacle
    to /usr/local/opt/carthage/Frameworks/CarthageKit.framework/Versions/A/Frameworks/Tentacle.framework/Versions/A/Tentacle
Error: Failed to fix install linkage
```

/cc @woodruffw

### CODE REVIEW

File: Formula/carthage.rb [hunk 1]
```diff
@@ -13,7 +13,12 @@ class Carthage < Formula
     sha256 "1d7ffb4d63ad048712d6b81f7e26fa4c5613d5dd6d34c3e5c46a086854d3a59e" => :sierra
   end
 
-  depends_on :xcode => ["9.0", :build]
+  depends_on :xcode => ["9.4", :build]
+
+  patch do
+    url "https://gist.githubusercontent.com/jdhealy/8db93b4e3791fa3715554b01e55d366d/raw/88d2913bd9cc0408a11a75715c593789ebcfb068/patch.diff"
```
[the remark this answers is not in the document]
Comment (ikesyo) [reply Formula/carthage.rb]: I think updating `Makefile` on upstream and release 0.31.1 will be appropriate or let's use `make swiftpm_install` instead.

[the remark this answers is not in the document]
Comment (jdhealy) [reply Formula/carthage.rb]: I guess I envisioned this under `While patches should generally be avoided, sometimes they are temporarily necessary` — then, definitely updating Makefile on the upstream and removing the `patch` from the formula in the PR for the 0.31.1 release. Would that be amenable?

I’ve heard from some who use older formulas when [bisecting](https://git-scm.com/docs/git-bisect) that having a working formula for this release would be nice.

[the remark this answers is not in the document]
Comment (fxcoudert) [reply Formula/carthage.rb]: 👍  for `inreplace`
Carthage is currently the most-downloaded formula that still cannot build on Mojave

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @jdhealy for the pull request!

**Pull request closed without merging**